src/main/java/org/killbill/billing/plugin/vertex/VertexTaxCalculator.java

```diff
@@ -305,6 +313,56 @@ private Collection<InvoiceItem> toInvoiceItems(final UUID invoiceId,
+            return taxItem;
+        }
+
+        final String taxItemDetails = createTaxItemDetails(taxRate);
+        if (taxItemDetails == null) {
+            return taxItem;
+        }
+
+        return new PluginInvoiceItem(new Builder<>()
+                                             .withId(taxItem.getId())
+                                             .withInvoiceItemType(taxItem.getInvoiceItemType())
+                                             .withInvoiceId(taxItem.getInvoiceId())
+                                             .withAccountId(taxItem.getAccountId())
+                                             .withChildAccountId(taxItem.getChildAccountId())
+                                             .withStartDate(taxItem.getStartDate())
+                                             .withEndDate(taxItem.getEndDate())
+                                             .withAmount(taxItem.getAmount())
+                                             .withCurrency(taxItem.getCurrency())
+                                             .withDescription(taxItem.getDescription())
+                                             .withSubscriptionId(taxItem.getSubscriptionId())
+                                             .withBundleId(taxItem.getBundleId())
+                                             .withCatalogEffectiveDate(taxItem.getCatalogEffectiveDate())
+                                             .withProductName(taxItem.getProductName())
+                                             .withPrettyProductName(taxItem.getPrettyProductName())
+                                             .withPlanName(taxItem.getPlanName())
+                                             .withPrettyPlanName(taxItem.getPrettyPlanName())
+                                             .withPhaseName(taxItem.getPhaseName())
+                                             .withPrettyPhaseName(taxItem.getPrettyPhaseName())
+                                             .withRate(taxItem.getRate())
+                                             .withLinkedItemId(taxItem.getLinkedItemId())
+                                             .withUsageName(taxItem.getUsageName())
+                                             .withPrettyUsageName(taxItem.getPrettyUsageName())
+                                             .withQuantity(taxItem.getQuantity())
+                                             .withItemDetails(taxItemDetails)
+                                             .withCreatedDate(taxItem.getCreatedDate())
+                                             .withUpdatedDate(taxItem.getUpdatedDate())
+                                             .validate().build());
+    }
+
+    private String createTaxItemDetails(@Nonnull final Double effectiveRate) {
+        try {
+            return objectMapper.writeValueAsString(ImmutableMap.of("taxRate", String.valueOf(effectiveRate)));
+        } catch (JsonProcessingException ignored) {
+            return null;
+        }
+    }
```

Comment:
can we make this generic in the sense  like `private String appendItemDetails(key, value)` and let caller decide key as well.

Reply:
I didn't get append thing, append to what?

Reply:
updated to be generic by passing the map with the details into the method